### PR TITLE
Don't announce blocks in `sync_to_tip_when_we_sync_together_with_multiple_peers`

### PR DESCRIPTION
**Pull request by altonen**
Don't spam block announcements and verify that all forks are downloaded by all peers.

Related to #12735

### CODE REVIEW

File: client/network/test/src/sync.rs [hunk 1]
```diff
@@ -818,20 +820,25 @@ fn sync_to_tip_when_we_sync_together_with_multiple_peers() {
...
 	net.block_until_idle();
 
-	assert!(!net.peer(2).has_block(block_hash));
-
-	net.peer(0).network_service().new_best_block_imported(block_hash, 10_000);
 	while !net.peer(2).has_block(block_hash) && !net.peer(1).has_block(block_hash) {
 		net.block_until_idle();
 	}
+
+	// verify that peer1 and peer3 also downloaded the other fork
```
Comment (lexnv) [inline client/network/test/src/sync.rs:839]:
```suggestion
	// verify that peer1 and peer3 also download the other fork
```

**Review by lexnv — APPROVED** [review]

File: client/network/test/src/sync.rs [hunk 2]
```diff
@@ -818,20 +820,25 @@ fn sync_to_tip_when_we_sync_together_with_multiple_peers() {
...
-			.push_blocks_at_without_informing_sync(BlockId::Number(0), 10_000, false);
+			.push_blocks_at_without_informing_sync(BlockId::Number(0), 10_000, false, false);
+
+	let block_hash2 =
+		net.peer(1)
+			.push_blocks_at_without_informing_sync(BlockId::Number(0), 5_000, false, false);
 
-	net.peer(1)
-		.push_blocks_at_without_informing_sync(BlockId::Number(0), 5_000, false);
+	assert!(!net.peer(2).has_block(block_hash));
```
Comment (bkchr) [inline client/network/test/src/sync.rs:829]: This changes the behavior of the test. I would like to keep the old order.

File: client/network/test/src/sync.rs [hunk 3]
```diff
@@ -818,20 +820,25 @@ fn sync_to_tip_when_we_sync_together_with_multiple_peers() {
...
 
-	assert!(!net.peer(2).has_block(block_hash));
-
-	net.peer(0).network_service().new_best_block_imported(block_hash, 10_000);
 	while !net.peer(2).has_block(block_hash) && !net.peer(1).has_block(block_hash) {
 		net.block_until_idle();
 	}
+
+	// verify that peer1 and peer3 also downloaded the other fork
+	assert!(net.peer(0).has_block(block_hash2));
```
Comment (bkchr) [inline client/network/test/src/sync.rs:840]: block_hash2 is not a fork. It is part of the same chain as block_hash

**Review by bkchr — APPROVED** [review]

File: client/network/test/src/sync.rs (hunk 3 above)
Comment (altonen) [reply client/network/test/src/sync.rs:840]: You're right. I confused myself about how `push_block_without_informing_sync_works`

**Pull request merged**